AppCenter/AppCenter/Internals/Channel/MSChannelUnitDefault.m

```diff
@@ -286,6 +292,8 @@ - (void)flushQueue {
                               self.pendingBatchQueueFull = NO;
                               if (self.availableBatchFromStorage) {
                                 [self flushQueue];
+                              } else {
+                                [[MSAuthTokenContext sharedInstance].storage removeAuthToken:tokenInfo.authToken];
```

Comment:
Add test for this logic

Reply:
This is definitely an edge case but what if an application is killed (or crashed) right before this line or it failed to remove auth token somehow? I think we need this clean up code somewhere else as well.

Reply:
In other words, how do we clean up auth token that's in the history but we already flushed all logs for the auth token?